### ISSUE TITLE
doubles().fromZeroToOne() generates values mostly around 0

### ISSUE THREAD

**Issue by marcin-mizianty** [NCR-CoDE/QuickTheories#16]
I'm not sure if that's intentional but the way QuickTheory is generating doubles isn't uniform. I know that `fromZeroToOne()` is weighted and it should generate a few 0s and 1s, but the problem is that it mostly generates really small numbers. 

The reason for that is using `Longs.range(0, ONE_CORRESPONDING_LONG).(…).as(i -> Double.longBitsToDouble(i), ...)`  where `ONE_CORRESPONDING_LONG = 4607182418800017408L` to generate doubles. The issue with that approach is that bits are not distributed uniformly among real domain, i.e. `Double.longBitsToDouble((long)(0.9*ONE_CORRESPONDING_LONG))` gives us `1.676329423594695E-31` and that is nowhere near 0.9 that you would expect. 

Is it intentional behaviour? If it is, could that be documented somewhere? Could uniform distribution be added to `doubles()` (could be called `uniformZeroToOne()` or `probabilities()`)?

**Comment by hcoles** [comment]
Hi @marcin-mizianty, thanks for the report.

No, this is not intentional.

I think ideally we'd extract something so that the distribution produced can be varied, but default it to something that gives a uniform distribution. But to start with making the behaviour less surprising would be sensible.

I'm unlikely to be able to look at this myself for a few weeks, but would be happy to accept a pull request if this is blocking you.

Thanks

Henry

**Comment by marcin-mizianty** [comment]
Thanks for your response. I have a work around (below) so I'm not blocked. However, I'll take a look at `doubles()` to see if I can come up with some better solution.

```java
public static final Source<Double> probabilities = Source.of((prng, step) ->
            (double) prng.nextLongWithinCheckedInterval(0, Long.MAX_VALUE) / Long.MAX_VALUE);
```

**Comment by philandstuff** [comment]
fixed by #17?

**Comment by marcin-mizianty** [comment]
Yes, I believe so.
